core/iwasm/common/wasm_c_api.c

```diff
@@ -320,6 +320,7 @@ wasm_engine_new()
         singleton_engine =
             wasm_engine_new_internal(Alloc_With_System_Allocator, NULL);
     }
+    singleton_engine->ref_count++;
```

Comment:
Hi, currently wasm-c-api APIs are not thread-safe, if to make them thread-safe, there might be more work to do. If you insist on it, could you please add a global lock, initialize it and lock/unlock it in the operations related to singleton:
after L314, define: `static korp_mutex c_api_global_lock;`
and here: use `os_mutex_init(&c_api_global_lock)` to initialize it, and check return value

We will merge #1010 to ensure the vector related operations be safe-thread, but if there is other issue occurring, we need to fix it.